---
title: "Phylogeographic inference with microtellite and SNPs"
author: |
  | Miguel Camacho-Sanchez${\ast^1}$, Guillermo Velo-Antón$^{1}$, Ana Veríssimo$^1$, Silvia Carvalho$^2$
  | $^1$ CIBIO/InBIO, Centro de Investigação em Biodiversidade e Recursos Genéticos da Universidade do Porto, Vairão, Portugal
  | Correspondence should be addressed to miguelcamachosachez@gmail.com
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  word_document: default
  pdf_document: default
---

```{r, include = FALSE}
knitr::opts_knit$set(root.dir = normalizePath("../.."))
library(dplyr)
```

```{r, include = FALSE}
# data
gen <- readRDS("data/intermediate/gen_consolidated_filtered.rds")
samples <- readRDS("data/intermediate/table1.rds")

#Results
#number of individuals H. molleri
indH <- names(gen) %>% grep(pattern = "hyla") %>%
            sapply(function(x) adegenet::nInd(gen[[x]])) %>% sum
#number of individuals P. cultripes
indP <- names(gen) %>% grep(pattern = "pelo") %>%
            sapply(function(x) adegenet::nInd(gen[[x]])) %>% sum
indHm <- adegenet::nInd(gen$usat_hyla)
indHs <- adegenet::nInd(gen$dart_hyla)
indPm <- adegenet::nInd(gen$usat_pelo)
indPs <- adegenet::nInd(gen$dart_pelo)
#total individuals
totalind <- indP + indH
#localities for H. molleri
popH <- samples$samples %>% grep(pattern = "H.m.") %>% length
#localities for P. cultripes
popP <- samples$samples %>% grep(pattern = "P.c.") %>% length
#missing data microtellite
# H. molleri
missHm <- hierfstat::genind2hierfstat(gen$usat_hyla) %>%
            dplyr::select(-1) %>% {xx <<- length(as.matrix(.)); . } %>%
            is.na() %>% {sum(.) / xx * 100} %>% round(1)
# P. cultripes
missPm <- hierfstat::genind2hierfstat(gen$usat_pelo) %>%
            dplyr::select(-1) %>% {xx <<- length(as.matrix(.)); . } %>%
            is.na() %>% {sum(.) / xx * 100} %>% round(1)
#missing data SNPs
missHs <- as.matrix(gen$dart_hyla) %>%
            {xx <<- length(.); . } %>%
            is.na() %>% {sum(.) / xx * 100} %>% round(1)
missPs <- as.matrix(gen$dart_pelo) %>%
            {xx <<- length(.); . } %>%
            is.na() %>% {sum(.) / xx * 100} %>% round(1)
#Number of loci
# H. molleri
#  microtellites
locHm <- adegenet::nLoc(gen$usat_hyla)
#  SNPs
locHs <- adegenet::nLoc(gen$dart_hyla)
# P. cultripes
#  microtellites
locPm <- adegenet::nLoc(gen$usat_pelo)
#  SNPs
locPs <- adegenet::nLoc(gen$dart_pelo)

#raw raw_genotypes
genraw <- readRDS("data/intermediate/raw_genotypes.rds")
rawHm <- adegenet::nInd(genraw$usat_hyla)
rawHs <- adegenet::nInd(genraw$dart_hyla)
rawPm <- adegenet::nInd(genraw$usat_pelo)
rawPs <- adegenet::nInd(genraw$dart_pelo)

#release memory
rm(genraw, gen)

```
# Manuscript information

Format: add journal title  \newline
Abstract word count: xx \newline
Main word count: xx  \newline
Number of references: xx  \newline
Number of figures: xx  \newline
Number of tables: xx  \newline
Number of text boxes: xx  \newline
Number of equations: xx  \newline
Keywords: microtellites, DArTseq, SNPs, neighbour joining, structure, phylogeography, Hyla molleri, Pelobates cultripes, Iberian Peninsula   \newline
Spelling: British English (en-GB)   \newline
\clearpage

# Abstract


\clearpage

# 1. Introduction

Microsatellites furnished at the 90's a powerful tool in ecology and conservation biology to assess variation at nuclear loci for the study of biological processes at the population level (Jehleh and Artzen 2002; Allendorf 2016). Their popularity has remained constant until present, despite the irruption and continuous growth in the use of SNPs since the 2000's (Putman and Carbone 2014; Allendorf 2016). Mutation rates in microsatellites are several orders of magnitude higher compared to SNPs across taxa (Dallas 1992; Weber and Wong 1993; Zhang & Hewitt, 2003; Lynch 2010). The high mutation rates combined with the large number of different a alleles for a single locus generate immense levels of polymorphism useful in population genetics (Avise 2004; Allendorf 2016). An immediate consequence is that microtellites are claimed to be more sentitive to sudden demographical processes (Morin 2004), so they could have more power to detect subtle population structure or recent bottlenecks (Luikart et al. 1998; Putman and Carbone 2014; Haasl and Payseur 2010). However, these premises have been questioned (Peery et al. 2012). High polymorphism comes together with high homoplasy (Garza and Freimer 1996; Hedrick 1999) and the difficulties to fit complex evolutionary models (Valdes et al. 1993; Rienzo et al 1994). The precedent can lead to estimate unreliable genetic distances for large divergence times (Kalinowski 2002), underestimate differentiation between populations caused with high intra-populational heterozigosity or overestimation of divergence caused by recent bottlenecks (Hedrick 1999). The simple evolutionary models in SNPs, high abundance and widespread distribution across the genome promised to overcome the limitations of microtellites in population genetics (Morin 2004). But, a microsatellite locus contains from 4 to 12 times more information than a SNPs (Liu et al. 2005), and the high initial costs of SNP discovery and sequencing (Morin 2004) constrained the use of SNPs to tens in non-model species or to thousands only in model organisms and humans (Liu et al. 2005; Pucket 2017). The first studies comparing few microsatellites to tens of SNPs found that the performance between both markers varied from study system and questions addressed (Puckett 2017), but that often, the few informative SNPs yielded the same amount of information than the microsatellite dataset (Liu et al. 2005; Narum et al. 2008; Hess et al. 2011). The development of Genotyping-by-Sequencing techniques, such as RADseq (Miller et al. 2007; Baird et al. 2008) and other similar techniques of reduction of genome complexity, coupled with the cost reduction of massive parallel sequencing has extended the reach of massive SNP genotyping to non-model organisms (Baird et al. 2008; Davey et al. 2011; Peterson et al. 2012). This has boosted the studies with thousands of SNPs in wildlife organism in conservation and evolutionary biology (Putman and Carbone 2014; Allendorf 2016; Andrews et al. 2016). The lesser per-locus information in SNPs is largely compensanted by the sequencing of thousands of them at the same cost than for a handful of microsatellites (Hodel et al. 2017; Puckett 2017). In the current genomic era, studies on wildlife organisms using thousands of SNPs typically find they being more powerful than microsatellites (Puckett 2017). So, yielding more defined clustering in non-parametric clustering (eg. PCA or DAPC; Rasic et al. 2014; Candy et al. 2015; Malenfant et al. 2015; Elbers et al. 2016; Jeffries et al. 2016; Puckett and Eggert 2016; Hodel et al. 2017), higher resolution to detect hybridization (Bradbury et al. 2015), better explanation for patterns of isolation by distance (Jeffries et al. 2016), more accuracy for estimating relatedness or origin of individuals (Puckett and Eggert 2016; Thrasner et al. 2018; Lemopoulos et al. 2019) or better trees (Bruneaux et al. 2012; Candy et al. 2015).
The above-mentioned sharp clustering with large-scale SNP data compared to microtellites using non-parametric methods does not reflect when using model-based clustering in the popular software STRUCTURE. This software provides a powerful tool to estimate individual ancestries from genotype data (Pritchard et al. 2000). The modeling of allele frequencies can be more powerful (Nassir et al. 2008), but comes at a high computer cost compared to non-parametric clustering (Porras-Hurtado et al. 2013), and in many cases, methods such as discriminant analysis of principal components claim to perform better (Jombart et al. 2008). Simulations show equivalent power of tens of microtellites and thousands of SNPs to retrieved with high confidence the correct number of cluster (Haasl and Payseur 2011). With empirical data, overall genetic structure with both datasets can vary in the number of most optimal number of clusters (best K) and proportions of individual ancestries (Bradbury et al. 2015; Malenfant et al. 2015; Elbers et al. 2016; Bohling et al. 2019) or be practically identical (Lemopoulos et al. 2019). However, the best K can be very sensitive to demographic processes and can vary substantially depending on the method to estimate it (Janes et al. 2017; Wang 2019). Other issues as uneven sampling and hierarchical population structure can also difficult the inference of the correct population structure (Puechmaille et al. 2016; Wang 2016). Neighbor joining clustering is a complementary way to overcome some limitations of model-based clustering and to recover a reliable evolutionary history of the populations under a wide variety of scenarios (Kalinowski 2009, 2011). The reconstruction of phylogenetic trees using neighbor joining with microtellite data are often coherent with the phylogeographic structure recovered with SNPs. However, the large homoplasy undermines their power to retrieve phylogenetic relationships between distant populations (Estoup et al. 2002). So, trees from microsatellite data recurrently have little support, specially at deeper nodes, which limits the interpretations of the evolutionary history of the populations (Moore et al. 2014; Candy et al. 2015; Bruneaux et al. 2012; Bradbury et al. 2018), although when several hundreds of loci are used the correct topology is recovered correctly (Takezaki and Nei 2008). The large amount of SNPs and their genome-wide distribution secure a range mutation rates which can provide sufficient information at different evolutionary scales, from recent within species processes to interspecies phylogenies (eg. Defaveri et al. 2013; Petersen et al. 2013).

We aimed to evaluate the phylogeographic patterns yielded by comparable datasets of microtellites and SNPs for two amphibians from the Iberian Peninsula, *Pelobates cultripes* and *Hyla molleri*. We hypothesize that the recent evolutionary window of microsatellites compared to the larger window in SNPs should lead to (1) model-based clustering for microsatellites yielding a higher number of clusters and less individual admixture and (2) very limited phylogeographic resolution of microtellite data to shallow evolutionary scales compared to a more robust phylogeographic resolution of SNPs at all scales. We used microtellite datasets from published studies, Gutiérrez-Rodríguez et al. (2017) (for *P. cultripes*) and Sánchez-Montes et al. (2019), for *H. molleri*, and newly sequenced panels of SNPs using DArTseq. Our tree reconstructions underline the strong limitations of typical microsatellite panels for phylogenetics, specially for getting information at deeper intra-specific processes, contrary to SNPs, which proved useful across all intra-species evolutionary scales.

# 2. Materials and Methods
## 2.1 Data collection
We used microsatellite and SNP datasets from two Iberian amphibians, *Hyla molleri* and *Pelobates cultripes*, covering most of their native range (Figure 1; Table S1).
The microsatellite genotypes were obtained from already-published studies. For *Hyla molleri*, we gathered them from *Appendix 1* in Sánchez-Montes et al. (2019). They consisted of `r rawHm` samples with `r locHm` loci. For *Pelobates cultripes*, we requested the genotypes to the authors in Gutiérrez-Rodríguez et al. (2017). This dataset consisted of `r rawPm` samples with `r locPm` loci.
SNP data was generated for `r rawHs` samples of *H. molleri* and `r rawPs` samples of *P. cultripes*. *???for Guillermo: I pasted the following from Gutiérrez-Rogríguez et al. (2017). Please, check: Most samples were obtained from tail tips of larvae and toes of adults, pre- served in absolute ethanol, and subsequently kept in a freezer at 20 °C???*. For *H. molleri*, we completed the sampling with tissue from several museum specimens (Table S1). Genomic DNA was extracted ExtractMe Genomic DNA 96-Well (DNA GDAŃSK), concentrated with QIAamp DNA Micro (QIAGEN GmbtH) kits, if necessary, and sequenced with DArTseq technology (Diversity Arrays Technology, Canberra, Australia).

## 2.2 Data generation
### 2.2.1 SNPs sequencing

Amphibians have large genomes with repetitive regions which hampers genomic sequencing approaches (Funk et al. 2018; Liedtke et al. 2018). We chose DArTseq, a proprietary protocol to sequence reduced representation of the genome from double-digested fragments. DArTseq has been claimed to work well with amphibians because it enriches for high-complexity regions, disregarding repetitive portions of the genome (Lambert et al. 2016). DNA extracts from *H. molleri* and *P. cultripes* with total DNA ranging from to xxx ug were sent to Diversity Arrays Technology (University of Canberra, Australia). The fragments were generated using the restriction enzymes xxx and xxx, and sequenced in an Illumina xxxx as sigle-end reads of 77 nucleotides (nt). The sequencing depth for *H. molleri* and *P. cultripes* were xxx and xxx M reads / sample, respectively. Diversity Arrays Technology genotypes the samples using the proprietary DArTSoft14 pipeline and returns a csv file with the genotypes and several parameters on each SNP. A xx % of the samples in the run are included as internal replicas to provide confidence levels on the genotype calls.

### 2.2.2 Data filtering
We additionally applied several filtering steps to the genotype matrices in R 3.5.1 (R Core Team 2018), using functions from the *dartR* 1.1.11 package (Gruber and Georges 2019) and custom functions (https://github.com/csmiguel/usat_snp; Figure S1). The genotypes were imported into R and filters were applied following order. A per-sample call rate threshold (proportion loci with calls) of 0.35, RepAvg (a repeatability measure of genotypes calls between replicas) threshold of 0.95. We used thresholds for allele balance of 0.15 and 0.85, and removed loci for which their coverage was 3.5 times higher than the median coverage (O'Leary et al. 2018), after which, we filtered loci with a call rate (proportion of samples with a call) below 0.8, retained only one SNP per contig, a set minor allele frequency threshold (MAF) of 0.02 and removed loci whose alleles deviated from Hardy-Weinberg proportions(O'Leary et al. 2018).
To make the datasets more comparable (Figure 1), we filtered out samples from spatially distant localities between marker datasets.

## 2.3 Population structure
We conducted model-based genetic structure analyses in STRUCTURE v2.3.4 (Pritchard et al. 2000). For each dataset, we performed 10 replicate runs from K = 1 to K = 8, since the best number of clusters fall within this range for both species (Gutiérrez-Rodríguez et al. 2017; Sánchez-Montes et al. 2019). We used an admixture model with correlated allele frequencies (Falush et al. 2003) and no prior information of sample origin. For the microtellite data, we used the same run lengths as in their original publications: 5*10^5 burnin followed by 1*10^6 iterations. For the SNP datasets, the run lengths were shorter, a burnin of 3*10^4 steps followed by 1 * 10^4 iterations, as for SNP data chain often converge fast (Pritchard, google groups...?). For the SNP data, we estimated lambda for K = 1, by averaging lambda estimates across 3 replicate runs. The values of lambda (0.60 for *H. molleri* and 0.69 for *P. cultripes*) were then used across all runs for the SNP data, whereas for microsatellites lambda was fixed to 1. These lower values of lambda can improve the modeling of correlated allele frequencies when using SNPs, where often the data is skewed toward rare alleles (Falush et al. 2003). STRUCTURE was run in parallel in 8 cores using Structure_threader (Pina-Martins et al. 2017), recording steps to log files every 50 and 5000 iterations for the SNP and microsatellite data, respectively. We run STRUCTURE analysis with random subsets of loci (n = 200, 500, 1000, 3000, 5000, 10000 and 20000) from the original matrix of SNPs for *P. cultripes* and the same STRUCTURE parameters. This approximation allowed to assess the effect of the number of loci in the patterns of genetic structure.
The convergence between the 10 replicate runs for each K was evaluated in R using Gelman and Rubin's convergence diagnostic (GR diagnostic; Gelman and Rubin 1992), with function *gelman.diag* in package *coda* (Plummer et al. 2006).
A central question in population genetics is determining the optimal number of populations subjacent to the genetic structure (Allendorf and Luikart 2007). However, hierarchical population structure (Evanno et al. 2005), unbalanced sampling and number markers (Wang 2017) can lead to the incorrect inference of number of populations. We used KFinder (Wang 2019) to compare the best number of populations for each dataset through 3 approaches: (1) Pr[X|K], the probability of data X given K populations (Pritchard et al. 2000) and, (2) ΔK, the most popular method and which considers the rate of change in the log probability of data between successive K (Evanno et al. 2005) and 3) PI, parsimony index, a newly proposed metric which favors K's which yield most consistent and with minimal average individual admixture, and which claims to be more consistent across wider demographic scenarios (Wang 2019).
We ran CLUMPAK (Kopelman et al. 2015) on STRUCTURE outputs. CLUMPAK feeds the software CLUMPP with the replicate runs at each value of K to generate consensus solutions for the distinct modes. It also computes the similarity between ancestry matrices from each run and it matches clusters across successive values of K.

## 2.4 Phylogenetic structure
We evaluated the phylogenetic relationships between the populations using the neighbor-joining (NJ) algorithm (Saitou and Nei 1987). We discarded using model-based phylogenetic inference for the implicit difficulties of fitting evolutionary and demographic models for intra-species trees (), and for the possibilities that offer comparing the 2 marker types (SNPs and microtellites) under a similar tree-reconstruction method. The application of NJ relies on a distance matrix computed from the genotypes. For the microtellite datasets we computed these matrices using Da distance (Nei et al. 1983) with *genet.dist* function from *hierfstat* 0.04-22 package (Goudet and Jombart 2015) in R. Da distances have been shown to yield more reliable topologies to other distances (Takezaki and Nei 1996, 2008). For the SNP genotypes, we computed euclidean distances with *stats::dist* function in R. Then NJ trees were reconstructed with *njs* function from *ape* 5.0 (Paradis and Schliep 2018) for each dataset. We computed the bootstrap support with a modified approach from *ape::boot.phylo* to accommodate non-standard input data. First, we resampled with replacement loci in the genotype matrices, from which we recomputed NJ trees. These trees were rooted using midpoint rooting with function *midpoint.root* from *phytools* 0.6-60 (Revell 2012) in R. And finally, bootstrap values were computed by counting the partitions of the original trees present in 1000 bootstrapped trees. We inspected the phylogeographic congruence between STRUCTURE and the phylogenetic by plotting together a the consensus solution for the major cluster from the output of CLUMPAK for K = 4, since K = 4, seemed to be an optimal solution for many of the datasets (Figure 2).
*Marker type and bootstrap support*. For each tree, branch lengths were normalized by dividing edge lengths by the length between the most distant tips in the tree. Then, we used several functions in *treeman* 1.1.3 package (Bennett 2018) in R, to extract 2 metrics from each node in the trees: 1) the distance from each node to the root (Dnr), and 2) the distance from each node to its parent node (Dnp). ??For the nodes in each tree we assessed the relation of Dnr and Dnp against their corresponding bootstrap support values fitting generalized linear models (GMLs) using a logit link function, with *stats::glm* function in R. *Actually the fit is really bad. See script 99_04_model_fitting.R and ask Jeff*??

*Number of markers and bootstrap support*. The number of SNPs can affect the phylogeographic patterns inferred from RAD-seq data (Hodel et al. 2017). We evaluated the effect of the number of SNPs on the support of the tree reconstructions. We sampled random subsets of 20000, 10000, 5000, 3000, 1000, 500 and 200 loci from the SNP dataset of *P. cultripes*. This process was repeated 5 times to overcome possible biases on marker choice. On each of these 5 subsets we computed 200 bootstrapped trees as above. We calculated the bootstrap supports by counting the number of partitions of the NJ tree from the full dataset (see above, `r adegenet::nLoc(gen$dart_pelo)` loci) the were present in the 1000 bootstrapped trees. The number of loci was linearized applying a natural logarithm and its relation with the bootstrap support was evaluated with a linear model in R.

*MAF and bootstrap support*. Loci can contain different amount of information. A way to quantify this information is by the MAF (ref). MAF thresholds used in filtering steps can greatly affect population genetic inferences, since allele frequencies are related to the mutation rate of their locus, thus  yielding information about different evolutionary scales (Linck and Battey 2019). Indeed, singletons blur the results of model-based genetic structure analysis (Linck and Battey 2019) and stringent MAF threshold (ie. 0.10) seem to yield best results (Haasl and Payseur 2011). However, their possible effect on the evolutionary relationships between populations has not been studied. We generated SNP datasets by applying MAF thresholds of 0.02, 0.04, 0.06, 0.08 and 0.10, to the SNPs from *P. cultripes*. From each dataset we sampled 500 random loci 5 times and computed bootstrap support from the resulting trees as above. We chose 500 loci because for this particular dataset it is a number of loci sensitive to changes on the overall bootstrap support of the tree (see Results). The relation between the bootstrap supports and MAF filters were assessed with a linear model in R.

# 3. RESULTS
After filtering (Figure S1), the dataset kept for analysis included a total `r totalind` individuals from `r nrow(samples)` localities: `r indH` individuals of *Hyla molleri* (microtellites: n = `r indHm`, `r locHm` loci, `r missHm` % missing data; SNPs: n = `r indHs`, `r locHs` loci, `r missHs` % missing data), from `r popH` localities, and `r indP` individuals of *Pelobates cultripes* (microtellites: n = `r indPm`, `r locPm` loci, `r missPm` % missing data; SNPs: n = `r indPs`, `r locPs` loci, `r missPs` % missing data), from `r popP` localities (Figure 1; Table S1).

## 3.1 Model-based clustering
*Convergence*. Convergence of the chains measured by the GR diagnostic varied between species and markers. Often, values below 1.05 indicate good convergence (Vats and Knudson 2018 Jeff should check this ref). For SNPs, the chains from each K in STRUCTURE converged well up to K = 2 in *H. molleri* and up to K = 6 in the case of *P. cultripes*. Reducing the number of loci did not affect the pattern of convergence across K in a consistent manner (Figures S2.1, S2.2, S2.5 - S2.11; Figure S3.1).
For microsatellite data, the chains converged consistently up to K = 6 in both species, although the GR diagnostic slightly peaked for K = 5 in *P. cultripes* (Figures S2.3 - S2.4; Figure S3.1).
*Optimal K*. The most optimal number of populations was evaluated with 3 metrics. Pr[X|K] increased steadily from K = 2 to K = 8 in all datasets (Figure 2 A-C). Evanno's ΔK peaked at K = 2 in all SNP datasets, followed by K = 3 (Figure 2 D-F). However, in the microsatellite dataset from *H. molleri*, ΔK peaked at K = 4 and K = 6 (Figure 2D). The predominant K = 2 in the SNP datasets were not in agreement with results from the Parsimony index (Figure 2 G-I), except for the microtellite data in *H. molleri*, in which PI and ΔK peaked at K = 4 (Figure 2G). For SNPs in *H. molleri*, the highest values of the PI were at K = 4 and K = 5, contrary to the K = 2 from ΔK (Figure 2G). In *P. cultripes*, the PI peaked at K = 4 for both markers, and then sunk in the SNP data, whereas for  microtellites the PI reached its maxima at K = 6 - K = 8 (Figure 2H). In the subsets of SNPs from *P. cultripes* the PI increased gradually up to K = 4. After that, the distinct marker sets showed inconsistent trends (Figure 2I).
??*clumpak table* talk about major/minor clusters in clumpak or similarities betweend ancestry matrices between mean population ancestries for usat/snp data or between n solutions for K??
For K = 2, the genetic structure was similar for microtellite and SNP data from both species (Figure 6), differentiating a norther and southern lineage for *H. molleri*, and a central-western and north-eastern linage for *P. cultripes*. However, from K = 3 to K = 8, the spatial patterns of genetic structure for both species showed moderate disagreements between SNPs versus microsatellites (Figure 6), which increased with higher values of K. These differences were more marked in *H. molleri*. For instance, in *H. molleri* the southeastern population of Ojos de Villaverde stood out as very distinct from K = 4 from SNPs whereas microtellites detected this singularity at K = 5.??should I focus on the biological patterns to allow a discussion on the phylogegrphy for these species or just on the marker comparison??

## 3.2 Phylogenetic structure
*Coherence between model-based and NJ clustering*. The phylogeographic structure of the model-based ancestries for the SNP dataset were coherent with the topology of their phylogenetic trees (Figure 4 A-B). That is, the samples from the same populations clustered together on the same or close clades and the ancestries from the samples were coherent with the phylogenetic relationships at different depths in the tree. However, the topology of the trees for microtellites were little supported. Furthermore, it neither provided a solid grouping of the samples from the same populations and it was not able to explain the relationships at different depths in the trees (Figure 4 C-D).??look at where the pops are located and talk about the spatial coherence of the phylogegraphic patterns. what can this info provides us regarding the phylogeography of these species respect to microtellites??

*Marker type and bootstrap support*. The topology of the trees from SNP data were highly supported compared to the trees from microtellites (Figures 3 and 4). The overall support was higher for the SNP dataset of *P. cultripes* than for *H. molleri* (Figure 3 A-B; Figure A-B). In microtellite data, larger distances of a node to the root (Dnr) were strongly correlated with higher bootstrap support (??statistics??)(Figure 3 C-D; Figure 4 C-D). This correlation was ¿¿not?? significant for the SNP data (??statistics??)(Figure 3 A-B; Figure 4 A-B). The distance of a node its precedent in direction to the root, or parent node (Dnp), explained a huge variation of in the bootstrap support for all datasets ??statistics?? (Figure 3 E-I; Figure 4 C-D). In the case of the SNP data Dnp ranged from 0? to 0.15?, but almost all trees supported nodes that had a Dnp above 0.025?. The slope and R^2 of the logarithmic regression for the SNP data were very large compared to the microsatellites (??statistics??; Figure 3 E-I). In *P. cultripes*, where the number of SNPs in the dataset was larger than in *H. molleri*, the slope was greater (statistics??; Figure 3 E-F)

*Number of markers and bootstrap support*. The bootstrap support was highest for the dataset with most loci (n = 33144), for which most nodes had a support equal to 1 (Figure 5). The bootstrap support decreased exponentially as the number of markers decreased down to 200 (Figure 5). The relation between bootstrap support and natural logarithm of number of loci was well explained by a linear model (R^2 = 0.34; p-value = < 0.001). The overall BS for the tree from the SNP dataset with lowest number of loci (n = 200 loci) was comparable to the BS for tree from the microtellite data of *H. molleri* (Figure 5).

*MAF and bootstrap support*. There was no effect of the MAF thresholds on the bootstrap support from the trees (t-test = 0.68, p-value = 0.5; Figure S4)


# 4 | DISCUSSION


# AUTHORS' CONTRIBUTIONS


# DATA ACCESSIBILITY



# ORCID

Miguel Camacho-Sanchez: 0000-0002-6385-7963 \newline
Guillermo Velo-Antón: 0000-0002-9483-5695 \newline
Ana Veríssimo: 0000-0003-3396-9822 \newline
Jeffrey O. Hanson: 0000-0002-4716-6134
Adam Marques: 0000-0002-6070-0506
Silvia B. Carvalho: 0000-0003-4368-4708 \newline

\clearpage

# REFERENCES
