---
title: "Phylogeographic inference with microtellite and SNPs"
author: |
  | Miguel Camacho-Sanchez${\ast^1}$, Guillermo Velo-Antón$^{1}$, Ana Veríssimo$^1$, Silvia Carvalho$^2$
  | $^1$ CIBIO/InBIO, Centro de Investigação em Biodiversidade e Recursos Genéticos da Universidade do Porto, Vairão, Portugal
  | Correspondence should be addressed to miguelcamachosachez@gmail.com
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  rmarkdown::pdf_document:
    toc: false
    keep_tex: true
    fig_caption: yes
    includes:
      in_header: preamble.tex
fontsize: 11pt
documentclass: article
subparagraph: true
---

```{r, include = FALSE}
knitr::opts_knit$set(root.dir = normalizePath("../.."))
```

```{r, include = FALSE}
# restore session
gen <- readRDS("data/intermediate/raw_genotypes.RDS")
# load packages
# define helper functions
# load data

```
# Manuscript information

Format: add journal title  \newline
Abstract word count: xx \newline
Main word count: xx  \newline
Number of references: xx  \newline
Number of figures: xx  \newline
Number of tables: xx  \newline
Number of text boxes: xx  \newline
Number of equations: xx  \newline
Keywords: microtellites, DArTseq, SNPs, neighbour joining, structure, phylogeography, Hyla molleri, Pelobates cultripes, Iberian Peninsula   \newline
Spelling: British English (en-GB)   \newline
\clearpage

# Abstract


\clearpage

# 1. Introduction
**general overview about the usat vs snp**
1-which are the characteristics of each marker
2-how informative are they
3-pros and cons for both
4-limitations
5-previous studies focusing on usats vs SNPs
**our case study**
1-study species: characteristics, distrigution, why them
2-markers datasets
**brief methodology with results and conclusions**


# 2. Materials and Methods
## 2.1 Data collection
Our analysis are built upon microsatellite and SNP datasets for two Iberian amphibians: *Hyla molleri* and *Pelobates cultripes*. For each species, the microtellite and SNP datasets come from different sample sets but covering both most of their native range (Figure 1).
The microsatellite genotypes were taken from already-published studies. For *Hyla molleri*, we gathered the microsatellite genotypes from *Appendix 1* in Sánchez-Montes et al. (2019). They consisted of `r adegenet::nInd(gen$usat_hyla)` samples with `r adegenet::nLoc(gen$usat_hyla)` genotyped microsatellites. For *Pelobates cultripes*, we requested the genotypes to the authors in Gutiérrez-Rodríguez et al. (2017). This dataset consisted of `r adegenet::nInd(gen$usat_pelo)` samples with `r adegenet::nLoc(gen$usat_pelo)` genotyped microsatellites.
The SNP data was generated using DArTseq technology (xx).
##

## Study area and species


## Data



# 3 | RESULTS

* result 1: sequencing
We sequenced a total of ´r adegenet::nInd(gen)´ samples which included 1 replica. DArTseq genotyped ´r adegenet::nLoc(gen)´SNPs in ´r nlevels(as.factor(gen@other$loc.metrics$CloneID))´ loci. The overall missing data was ´r ´ ()

* result 2: structure vs dapc

* result 3: genetic diversity


# 4 | DISCUSSION

* Overview

* Key result 1

* Key result 2

* Key result 3

* Concluding paragraph

# AUTHORS' CONTRIBUTIONS

TODO

# DATA ACCESSIBILITY



# ORCID

Miguel Camacho-Sanchez: 0000-0002-6385-7963 \newline
Guillermo Velo-Antón: 0000-0002-9483-5695 \newline
Ana Veríssimo: 0000-0003-3396-9822 \newline
Jeffrey O. Hanson: 0000-0002-4716-6134
Adam Marques: 0000-0002-6070-0506
Silvia B. Carvalho: 0000-0003-4368-4708 \newline

\clearpage

# REFERENCES
