---
title: "Phylogeographic inference with microtellite and SNPs"
author: |
  | Miguel Camacho-Sanchez${\ast^1}$, Guillermo Velo-Antón$^{1}$, Ana Veríssimo$^1$, Silvia Carvalho$^2$
  | $^1$ CIBIO/InBIO, Centro de Investigação em Biodiversidade e Recursos Genéticos da Universidade do Porto, Vairão, Portugal
  | Correspondence should be addressed to miguelcamachosachez@gmail.com
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  rmarkdown::pdf_document:
    toc: false
    keep_tex: true
    fig_caption: yes
    includes:
      in_header: preamble.tex
fontsize: 11pt
documentclass: article
subparagraph: true
---

```{r, include = FALSE}
knitr::opts_knit$set(root.dir = normalizePath("../.."))
```

```{r, include = FALSE}
# restore session
gen <- readRDS("data/intermediate/raw_genotypes.RDS")
# load packages
# define helper functions
# load data

```
# Manuscript information

Format: add journal title  \newline
Abstract word count: xx \newline
Main word count: xx  \newline
Number of references: xx  \newline
Number of figures: xx  \newline
Number of tables: xx  \newline
Number of text boxes: xx  \newline
Number of equations: xx  \newline
Keywords: microtellites, DArTseq, SNPs, neighbour joining, structure, phylogeography, Hyla molleri, Pelobates cultripes, Iberian Peninsula   \newline
Spelling: British English (en-GB)   \newline
\clearpage

# Abstract


\clearpage

# 1. Introduction

Microsatellites furnished at the 90's a powerful tool in ecology and conservation biology to assess variation at nuclear loci for the study of biological processes at the population level (Jehleh and Artzen 2002; Allendorf 2016). Their popularity has remained constant until present, despite the irruption and continuous growth in the use of SNPs since the 2000's (Putman and Carbone 2014; Allendorf 2016). Mutation rates in microsatellites are several orders of magnitude higher compared to SNPs across taxa (Dallas 1992; Weber and Wong 1993; Zhang & Hewitt, 2003; Lynch 2010). The high mutation rates combined with the large number of different a alleles for a single locus generate immense levels of polymorphism useful in population genetics (Avise 2004; Allendorf 2016). An immediate consequence is that microtellites are claimed to be more sentitive to sudden demographical processes (Morin 2004), so they could have more power to detect subtle population structure or recent bottlenecks (Luikart et al. 1998; Putman and Carbone 2014; Haasl and Payseur 2010). However, these premises have been questioned (Peery et al. 2012). High polymorphism comes together with high homoplasy (Garza and Freimer 1996; Hedrick 1999) and the difficulties to fit complex evolutionary models (Valdes et al. 1993; Rienzo et al 1994). The precedent can lead to estimate unreliable genetic distances for large divergence times (Kalinowski 2002), underestimate differentiation between populations caused with high intra-populational heterozigosity or overestimation of divergence caused by recent bottlenecks (Hedrick 1999). The simple evolutionary models in SNPs, high abundance and widespread distribution across the genome promised to overcome the limitations of microtellites in population genetics (Morin 2004). But, a microsatellite locus contains from 4 to 12 times more information than a SNPs (Liu et al. 2005), and the high initial costs of SNP discovery and sequencing (Morin 2004) constrained the use of SNPs to tens in non-model species or to thousands only in model organisms and humans (Liu et al. 2005; Pucket 2017). The first studies comparing few microsatellites to tens of SNPs found that the performance between both markers varied from study system and questions addressed (Puckett 2017), but that often, the few informative SNPs yielded the same amount of information than the microsatellite dataset (Liu et al. 2005; Narum et al. 2008; Hess et al. 2011). The development of Genotyping-by-Sequencing techniques, such as RADseq (Miller et al. 2007; Baird et al. 2008) and other similar techniques of reduction of genome complexity, coupled with the cost reduction of massive parallel sequencing has extended the reach of massive SNP genotyping to non-model organisms (Baird et al. 2008; Davey et al. 2011; Peterson et al. 2012). This has boosted the studies with thousands of SNPs in wildlife organism in conservation and evolutionary biology (Putman and Carbone 2014; Allendorf 2016; Andrews et al. 2016). The lesser per-locus information in SNPs is largely compensanted by the sequencing of thousands of them at the same cost than for a handful of microsatellites (Hodel et al. 2017; Puckett 2017). In the current genomic era, studies on wildlife organisms using thousands of SNPs typically find they being more powerful than microsatellites (Puckett 2017). So, yielding more defined clustering in non-parametric clustering (eg. PCA or DAPC; Rasic et al. 2014; Candy et al. 2015; Malenfant et al. 2015; Elbers et al. 2016; Jeffries et al. 2016; Puckett and Eggert 2016; Hodel et al. 2017), higher resolution to detect hybridization (Bradbury et al. 2015), better explanation for patterns of isolation by distance (Jeffries et al. 2016), more accuracy for estimating relatedness or origin of individuals (Puckett and Eggert 2016; Thrasner et al. 2018; Lemopoulos et al. 2019) or better trees (Bruneaux et al. 2012; Candy et al. 2015).
The above-mentioned sharp clustering with large-scale SNP data compared to microtellites using non-parametric methods does not reflect when using model-based clustering in the popular software STRUCTURE. This software provides a powerful tool to estimate individual ancestries from genotypes data (Pritchard et al. 2000). Simulations show equivalent power of tens of microtellites and thousands of SNPs to retrieved with high confidence the correct number of cluster (Haasl and Payseur 2011). With empirical data, overall genetic structure with both datasets can vary in the number of most optimal number of clusters (best K) and proportions of individual ancestries (Bradbury et al. 2015; Malenfant et al. 2015; Elbers et al. 2016; Bohling et al. 2019) or be practically identical (Lemopoulos et al. 2019). However, the best K can be very sensitive to demographic processes and can vary substantially depending on the method to estimate it (Janes et al. 2017; Wang 2019). Other issues as uneven sampling and hierarchical population structure can also difficult the inference of the correct population structure (Puechmaille et al. 2016; Wang 2016). Neighbor joining clustering is a complementary way to overcome some limitations of model-based clustering and to recover a reliable evolutionary history of the populations under a wide variaty of scenarios (Kalinowski 2009, 2011). The reconstructions of phylogenetic trees using neighbor joining with microtellite data are often coherent with the phylogeographic structure recovered with SNPs. However, the large homoplasy undermines their power to retrieve phylogenetic relationships between distant populations (Estoup et al. 2002). So, trees from microsatellite data recurrently have little support, specially at deeper nodes, which limits the interpretations of the evolutionary history of the populations (Moore et al. 2014; Candy et al. 2015; Bruneaux et al. 2012; Bradbury et al. 2018), although when several hundreds of loci are used the correct topology is recovered correctly (Takezaki and Nei 2006). The large amount of SNPs and their genome-wide distribution secure a range mutation rates which can provide sufficient information at different evolutionary scales, from recent within species processes to interspecies phylogenies (eg. Defaveri et al. 2013; Petersen et al. 2013).

We aimed to evaluate the phylogeographic patterns yielded by comparable datasets of microtellites and SNPs for two amphibians from the Iberian Peninsula, *Pelobates cultripes* and *Hyla molleri*. We hypothesize that the recent evolutionary window of microsatellites compared to the larger window in SNPs should lead to (1) model-based clustering for microsatellites yielding a higher number of clusters and less individual admixture and (2) very limited phylogeographic resolution of microtellite data to shallow evolutionary scales compared to a more robust phylogeographic resolution of SNPs at all scales. We used microtellite datasets from published studies, Gutiérrez-Rodríguez et al. (2017) (for *P. cultripes*) and Sánchez-Montes et al. (2019), for *H. molleri*, and newly sequenced panels of SNPs using DArTseq. Our tree reconstructions underline the strong limitations of typical microsatellite panels for phylogenetics, specially for getting information at deeper intra-specific processes, contrary to SNPs, which proved useful across all intra-species evolutionary scales.

# 2. Materials and Methods
## 2.1 Data collection
We used microsatellite and SNP datasets from two Iberian amphibians, *Hyla molleri* and *Pelobates cultripes*, covering most of their native range (Figure 1).
The microsatellite genotypes were obtained from already-published studies. For *Hyla molleri*, we gathered the microsatellite genotypes from *Appendix 1* in Sánchez-Montes et al. (2019). They consisted of `r adegenet::nInd(gen$usat_hyla)` samples with `r adegenet::nLoc(gen$usat_hyla)` genotyped microsatellites. For *Pelobates cultripes*, we requested the genotypes to the authors in Gutiérrez-Rodríguez et al. (2017). This dataset consisted of `r adegenet::nInd(gen$usat_pelo)` samples with `r adegenet::nLoc(gen$usat_pelo)` genotyped microsatellites.
The SNP data was generated using DArTseq technology (Diversity Arrays Technology, Canberra, Australia). We included samples spanning all over the Iberian Peninsula (Figure 1; Table 1). DNA was extracted ExtractMe Genomic DNA 96-Well (DNA GDAŃSK), concentrated with QIAamp DNA Micro (QIAGEN GmbtH) kits, if necessary, and sent for sequencing.
##

## Study area and species


## Data



# 3 | RESULTS

* result 1: sequencing
We sequenced a total of ´r adegenet::nInd(gen)´ samples which included 1 replica. DArTseq genotyped ´r adegenet::nLoc(gen)´SNPs in ´r nlevels(as.factor(gen@other$loc.metrics$CloneID))´ loci. The overall missing data was ´r ´ ()

* result 2: structure vs dapc

* result 3: genetic diversity


# 4 | DISCUSSION

* Overview

* Key result 1

* Key result 2

* Key result 3

* Concluding paragraph

# AUTHORS' CONTRIBUTIONS

TODO

# DATA ACCESSIBILITY



# ORCID

Miguel Camacho-Sanchez: 0000-0002-6385-7963 \newline
Guillermo Velo-Antón: 0000-0002-9483-5695 \newline
Ana Veríssimo: 0000-0003-3396-9822 \newline
Jeffrey O. Hanson: 0000-0002-4716-6134
Adam Marques: 0000-0002-6070-0506
Silvia B. Carvalho: 0000-0003-4368-4708 \newline

\clearpage

# REFERENCES
