---
title: "Phylogeographic inference with microtellite and SNPs"
author: |
  | Miguel Camacho-Sanchez${\ast^1}$, Guillermo Velo-Antón$^{1}$, Ana Veríssimo$^1$, Silvia Carvalho$^2$
  | $^1$ CIBIO/InBIO, Centro de Investigação em Biodiversidade e Recursos Genéticos da Universidade do Porto, Vairão, Portugal
  | Correspondence should be addressed to miguelcamachosachez@gmail.com
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  rmarkdown::pdf_document:
    toc: false
    keep_tex: true
    fig_caption: yes
    includes:
      in_header: preamble.tex
fontsize: 11pt
documentclass: article
subparagraph: true
---

```{r, include = FALSE}
knitr::opts_knit$set(root.dir = normalizePath("../.."))
```

```{r, include = FALSE}
# restore session
gen <- readRDS("data/intermediate/raw_genotypes.RDS")
# load packages
# define helper functions
# load data

```
# Manuscript information

Format: add journal title  \newline
Abstract word count: xx \newline
Main word count: xx  \newline
Number of references: xx  \newline
Number of figures: xx  \newline
Number of tables: xx  \newline
Number of text boxes: xx  \newline
Number of equations: xx  \newline
Keywords: microtellites, DArTseq, SNPs, neighbour joining, structure, phylogeography, Hyla molleri, Pelobates cultripes, Iberian Peninsula   \newline
Spelling: British English (en-GB)   \newline
\clearpage

# Abstract


\clearpage

# 1. Introduction
**general overview about the usat vs snp**
Genetic markers represent a central pillar upon which Conservation and Evolutionary Biology are built. Ideally, neutral genetic markers should be representative of genome-wide processes and have  adequate levels of information to address the desired demographic or evolutionary questions. Under these premises, the popularity of different types of neutral genetic markers have been oscillating since the advent of molecular biology. While short-sequence tandem repeats (microtellites hereafter) have been at the frontline of several biological disciplines since the past xx years, the rise of high-throughput sequencing in the last decade has pushed single nucleotide polymorphisms (SNPs) at this forefrontline as well, and, at present, these are two of the most widely used markers in xx research. Each have different characteristics deriving from their molecular nature. Microsatellites are repeats of short-motifs (xx -- bp) in the genome caused by slippage of the DNA-polymerase, whereas SNPs are the form of single nucleotide (ATCG). Due to the relatively high rate of polymerase slippage compared to base misincorporation, microtellites have mutation rates which are several orders of magnitude higher than SNPs. For instance, in vertebrates?? the mutation rate can vary from xx to xxx, whereas the background mutation rate for vertebrates genomes (often used for mutation rate in SNPs, assuming they are randomly distributed across the genome) is around 10-9. The number of alleles for a given microsatellite can be very high due to the presumably infinite number of repeats (assuming an infinite allele site model) it could adopt. However, SNPs are restricted to four different alleles corresponding to the 4 nucleotides (ATCG), and often only biallelic SNPs are considered in most studies.
Microtellites have been a preferable marker choice for population genetics since xxx years, since they claim to be more sensitive to demographic processes at shallow evolutionary scales due to their high levels of polymorphism and high mutation rates. On the contrary, SNPs contain little information less information per locus. The rise of high-throughput sequencing in the last decade has pushed SNPs to the frontline in population genetics. The lesser per-locus information in SNPs is compensated by the sequencing of thousands of them at the same cost than for microsatellites. Approximately, developeing and sequencing a panel of xxx microsatellites has the same cost than xxx SNPs (Hodel 2016??). Genome-wide SNPs secure a distribution of mutation rates, from low to high, which can provide information and different evolutionary scales from recent within species processes to interspecies, contrary to microsatellites for which homoplasies and the difficulties to fit an evolutionary model only make them suitable for whithin species-processes.

Whereas the value of microtellites has been largely proven for population genetics, the attemps to scale their information into a phylogenetics framework (ie phylogeography) have had little success. Indeed, the literature is full of attempts to recover within species phylogenetic reconstructures which recurrently fall into the issue of being supported. Recent studies have tried to make comparisons on the levels of iformation returned for microsatelllites versus SNPs for the sampe study system. They find...... However, when referening to comparisons for the genetic structure, they often refer terms "resolution" based on visual examination of the results between different marker, but they do not provide any statistic. Also, they have not dealt with the issue of the information that they can provide in a phylogenetics framework.
We evaluated the genetic structure and phylogeography information yielded by comparable datasets of microtellites and SNPs for two amphibians from the Iberian Peninsula, *Pelobates cultripes* and *Hyla molleri*. ?? aims??? hypotheses?? We used microtellite datasets from published studies, Gutiérrez-Rodríguez et al. (2017) (for *P. cultripes*) and Sánchez-Montes et al. (2019), for *H. molleri*, and newly sequenced panels of SNPs. Our tree reconstructions underline the strong limitations of using microsatellites for phylogenetics, specially for getting information at deeper intra-specific processes, contrary to SNPs, which proved useful across all depths in the trees. For population structure xxxxx.

# 2. Materials and Methods
## 2.1 Data collection
Our analysis are built upon microsatellite and SNP datasets for two Iberian amphibians: *Hyla molleri* and *Pelobates cultripes*. For each species, the microtellite and SNP datasets come from different sample sets but covering both most of their native range (Figure 1).
The microsatellite genotypes were taken from already-published studies. For *Hyla molleri*, we gathered the microsatellite genotypes from *Appendix 1* in Sánchez-Montes et al. (2019). They consisted of `r adegenet::nInd(gen$usat_hyla)` samples with `r adegenet::nLoc(gen$usat_hyla)` genotyped microsatellites. For *Pelobates cultripes*, we requested the genotypes to the authors in Gutiérrez-Rodríguez et al. (2017). This dataset consisted of `r adegenet::nInd(gen$usat_pelo)` samples with `r adegenet::nLoc(gen$usat_pelo)` genotyped microsatellites.
The SNP data was generated using DArTseq technology (xx).
##

## Study area and species


## Data



# 3 | RESULTS

* result 1: sequencing
We sequenced a total of ´r adegenet::nInd(gen)´ samples which included 1 replica. DArTseq genotyped ´r adegenet::nLoc(gen)´SNPs in ´r nlevels(as.factor(gen@other$loc.metrics$CloneID))´ loci. The overall missing data was ´r ´ ()

* result 2: structure vs dapc

* result 3: genetic diversity


# 4 | DISCUSSION

* Overview

* Key result 1

* Key result 2

* Key result 3

* Concluding paragraph

# AUTHORS' CONTRIBUTIONS

TODO

# DATA ACCESSIBILITY



# ORCID

Miguel Camacho-Sanchez: 0000-0002-6385-7963 \newline
Guillermo Velo-Antón: 0000-0002-9483-5695 \newline
Ana Veríssimo: 0000-0003-3396-9822 \newline
Jeffrey O. Hanson: 0000-0002-4716-6134
Adam Marques: 0000-0002-6070-0506
Silvia B. Carvalho: 0000-0003-4368-4708 \newline

\clearpage

# REFERENCES
